src/lib/archwizard.module.ts

```diff
@@ -62,7 +69,12 @@ import {WizardStepDirective} from './directives/wizard-step.directive';
 })
 export class ArchwizardModule {
   /* istanbul ignore next */
-  static forRoot(): ModuleWithProviders {
-    return {ngModule: ArchwizardModule, providers: []};
+  static forRoot(config?: ArchwizardModuleConfig): ModuleWithProviders {
+    return {
+      ngModule: ArchwizardModule,
+      providers: [
+        { provide: NAVIGATION_MODE_FACTORY, useClass: config && config.navigationModeFactory || BaseNavigationModeFactory },
```

Comment:
`config && config.navigationModeFactory` allows the library user to overwrite the default factory (i.e. `BaseNavigationModeFactory`), right?

Reply:
Yes, this construction makes sure that user's setting only takes effect when `config` is not falsy and `config.navigationModeFactory` is not falsy. Otherwise `BaseNavigationModeFactory` will be used.

This expressioncan be rewritten as:
```
config && config.navigationModeFactory ? config.navigationModeFactory : BaseNavigationModeFactory
```
which is a little bit longer.